include/irr/asset/IAsset.h

```diff
@@ -11,13 +11,20 @@ namespace asset
 
 class IAssetManager;
 
+//! A class managing Asset's extra metadata context
+/**
+	Sometimes there may be nedeed attaching some metadata by a Loader
+	into Asset structure - that's why the class is defined.
+	Pay attention that it hasn't been done exactly yet, engine doesn't provide
+	metadata injecting.
+*/
 class IAssetMetadata : public core::IReferenceCounted
 {
 protected:
     virtual ~IAssetMetadata() = default;
 
 public:
-    //! this could actually be reworked to something more usable
+    //! This could actually be reworked to something more usable
```

Comment:
because of external and custom asset loaders we cannot protect a static_cast with a type enum comparison.. this is why we provide a string.

Reply:
status?

Reply:
Solved